distributed-job-manager/src/test/kotlin/ru/fix/distributed/job/manager/strategy/ReassignmentNumberComparisonTest.kt

```diff
@@ -0,0 +1,700 @@
+            "worker-0"{
+                "job-1"(
+                        "work-item-0",
+                        "work-item-1"
+                )
+            }
+            "worker-1"{
+                "job-1"(
+                        "work-item-2",
+                        "work-item-3"
+                )
+            }
+            "worker-2"{
+                "job-1"(
+                        "work-item-4",
+                        "work-item-5"
+                )
+            }
+        }
+
+        val results = reassignmentResults(available, previous)
+        assertEquals(2, results.evenlySpreadReassignmentNumber)
+        assertEquals(4, results.rendezvousReassignmentNumber)
+    }
+
+    @Test
+    fun `balance items when three workers was alive and one worker destroyed`() {
+        val workPool: JobScope.() -> Unit = {
+            "job-0"(
+                    "work-item-0",
+                    "work-item-1",
+                    "work-item-2"
+            )
+            "job-1"(
+                    "work-item-3",
+                    "work-item-4",
+                    "work-item-5",
+                    "work-item-6"
+            )
+            "job-2"(
+                    "work-item-7"
+            )
+            "job-3"(
+                    "work-item-8",
+                    "work-item-9",
+                    "work-item-10"
+            )
+        }
+        val available = assignmentState {
+            "worker-0"(workPool)
+            "worker-1"(workPool)
+        }
+
+        val previous = assignmentState {
+            "worker-0" {
+                "job-0"(
+                        "work-item-0"
+                )
+                "job-1"(
+                        "work-item-3",
+                        "work-item-6"
+                )
+                "job-3"(
+                        "work-item-9"
+                )
+            }
+            "worker-1" {
+                "job-0"(
+                        "work-item-1"
+                )
+                "job-1"(
+                        "work-item-4"
+                )
+                "job-2"(
+                        "work-item-7"
+                )
+                "job-3"(
+                        "work-item-10"
+                )
+            }
+            "worker-2" {
+                "job-0"(
+                        "work-item-2"
+                )
+                "job-1"(
+                        "work-item-5"
+                )
+                "job-3"(
+                        "work-item-8"
+                )
+            }
+        }
+
+        val results = reassignmentResults(available, previous)
+        assertEquals(3, results.evenlySpreadReassignmentNumber)
+        assertEquals(10, results.rendezvousReassignmentNumber)
+    }
+
+    @Test
+    fun `reboot one worker`() {
+        val workPoolForSws: JobScope.() -> Unit = {
+            "rebill-job"(
+                    "rebill-item-0",
+                    "rebill-item-1",
```

Comment:
suggestion: we can take real life example from our working cluster here. 
Here is an example. Lets consider our prod sws servers: 
sws 8 servers in total
32 jobs with single work-item
4 jobs with 2 work-item each
5 jobs with 3 work-item each
2 jobs with 16 work-item each
1 job with 285 work-items

scenarios to run: 
1. rolling update: restart all servers  one by  one
2. rolling update: restart all servers  two by  two
3. shutdown all servers exept one and then launch all 8 servers again.

Reply:
done